### ISSUE TITLE
name id 4 (FULL_FONT_NAME) check nuance

### ISSUE THREAD

**Issue by davelab6** [googlefonts/fontbakery#1517]
Recently we changed name id 4 from "Calligraffitti" to "Calligraffitti Regular". Some systems use name id 4 to match fonts, so this change makes those systems see the font as a separate family.

OT spec for name id 4 says,

> Full font name that reflects all family and relevant subfamily descriptors. The full font name is generally a combination of name IDs 1 and 2, or of name IDs 16 and 17, or a similar human-readable variant. 
> 
> For fonts in extended typographic families (that is, families that include more than regular, italic, bold, and bold italic variants), values for name IDs 1 and 2 are normally chosen to provide compatibility with certain applications that assume a family has at most four style-linked fonts. In that case, some fonts may end up with a Subfamily name (name ID 2) of “Regular” even though the font would not be considered, typographically, a regular font. **For such non-regular fonts in which name ID 2 is specified as “Regular”, the “Regular” descriptor would generally be omitted from name ID 4. For example, the Arial Black font has a Font Family name (name ID 1) of “Arial Black” and a Subfamily name (name ID 2) of “Regular”, but has a full font name (name ID 4) of “Arial Black”.**
>
> **Note that name IDs 16 and 17 should also be included in these fonts, and that name ID 4 would typically be a combination of name IDs 16 and 17, without needing any additional qualifications regarding “Regular”.**

Do our current checks ensure that `Regular` is only applied when the font is not actually a true regular?

**Comment by anthrotype** [comment]
wait... Glyphs.app always keeps "Regular" in the Full Name (name id 4) for the fonts that are "true" Regular (i.e. their preferred subfamily name is "Regular").

I recently changed ufo2ft to match that:
https://github.com/googlei18n/ufo2ft/pull/148/commits/bb6b92256c290861b15af328273a72683bf0e13d

I thought you wanted to match what Glyphs.app does?

**Comment by anthrotype** [comment]
See also https://github.com/googlei18n/fontmake/issues/209#issuecomment-321239770

**Comment by davelab6** [comment]
>> Do our current checks ensure that Regular is only applied when the font is not actually a true regular?
>
> Glyphs.app always keeps "Regular" in the Full Name (name id 4) for the fonts that are "true" Regular 

We are saying the same thing, I think :)

**Comment by anthrotype** [comment]
ah ok then! 😅

**Comment by felipesanches** [comment]
The current rule implemented on test/017 (validation of several name table entries) is that the expected value for FULL_FONT_NAME (nameid = 4) is:

```
"{} {}".format(fname_with_spaces, style_with_spaces)
```

with a special case that when the style is Regular it can be omitted but we emit a warning about it (see: https://github.com/googlefonts/fontbakery/issues/1436).

**Comment by felipesanches** [comment]
Now that I have finished splitting up test/017 into individual per-name-entry tests, the nameid 4 (FULL_FONT_NAME) validation is isolated in **com.google.fonts/test/159** - "Check name table: FULL_FONT_NAME entries."

https://github.com/googlefonts/fontbakery/blob/f416aa79af1a24782140f3aadc2f4409cd5c3342/Lib/fontbakery/specifications/googlefonts.py#L4299-L4338

**Comment by felipesanches** [comment]
The tests dealing with NAMEID-4 are:
* **test/068** "Does full font name begin with the font family name?"
* **test/071** "Font follows the family naming recommendations?"
* **test/094** "METADATA.pb 'fullname' value matches internal 'fullname' ?"
* **test/106** "METADATA.pb font.style 'italic' matches font internals ?"
* **test/107** "METADATA.pb font.style "normal" matches font internals ?"
* **test/108** "Metadata key-value match to table name fields?"
* **test/156** "Font has all mandatory 'name' table entries ?"
* **test/159** "Check name table: FULL_FONT_NAME entries."

**Comment by felipesanches** [comment]
I've run this selection of Font Bakery tests (from latest commit: https://github.com/googlefonts/fontbakery/commit/f416aa79af1a24782140f3aadc2f4409cd5c3342) on the current version of Calligraffitti from the google/fonts repo (https://github.com/google/fonts/commit/d2b711b5182699fd7d272b34424f753efdce69cd) and it all looks good:

![screenshot at 2017-08-30 01 33 19](https://user-images.githubusercontent.com/213676/29855711-33c3bb80-8d24-11e7-9842-4e9614984e78.png)

If you want to run it yourself, this was the command I used:
```
fontbakery check-googlefonts -c com.google.fonts/test/094 -c com.google.fonts/test/071 -c com.google.fonts/test/068 -c com.google.fonts/test/159 -c com.google.fonts/test/156 -c com.google.fonts/test/106 -c com.google.fonts/test/107 -c com.google.fonts/test/108 ~/gfonts/apache/calligraffitti/*.ttf --verbose
```

![screenshot at 2017-08-30 01 43 46](https://user-images.githubusercontent.com/213676/29855761-b0720ed4-8d24-11e7-93dc-a3649ad3bd9c.png)

**Comment by felipesanches** [comment]
@davelab6 Please review the above and let me know if you're happy with the current selection of NAME-ID 4 tests.

**Comment by felipesanches** [comment]
All tests that deal with the FULL_FONT_NAME (name ID = 4) entries now have automated self-tests on Travis. (I just implemented them on the #1561 PR) The self-tests contain examples for all possible scenarios. If any of these tests still produce undesirable results, we must review their corresponding self-tests to add the desired behaviour there and then it helps to assure the implementation adheres to that.

This gives me confidence to close this issue, since the current implementation passes the self-tests and they contain a full description of all requirements that were expressed here (and in other issues) up to now.
